public/app/core/components/RolePicker/RolePickerMenu.tsx

```diff
@@ -0,0 +1,585 @@
+    }
+    setSelectedOptions([]);
+  };
+
+  const onClearSubMenu = () => {
+    const options = selectedOptions.filter((role) => {
+      const groupName = getRoleGroup(role);
+      return groupName !== openedMenuGroup;
+    });
+    setSelectedOptions(options);
+  };
+
+  const onUpdateInternal = () => {
+    const selectedCustomRoles: string[] = [];
+    for (const key in selectedOptions) {
+      const roleUID = selectedOptions[key]?.uid;
+      selectedCustomRoles.push(roleUID);
+    }
+    onUpdate(selectedBuiltInRole, selectedCustomRoles);
+  };
+
+  return (
+    <div className={cx(styles.menu, customStyles.menuWrapper)}>
+      <div className={customStyles.menu} aria-label="Role picker menu">
+        <CustomScrollbar autoHide={false} autoHeightMax="300px" hideHorizontalTrack>
+          <div className={customStyles.menuSection}>
+            <div className={customStyles.groupHeader}>Built-in roles</div>
+            <RadioButtonGroup
+              className={customStyles.builtInRoleSelector}
+              options={BuiltinRoleOption}
+              value={selectedBuiltInRole}
+              onChange={onSelectedBuiltinRoleChange}
+              fullWidth={true}
+            />
+          </div>
+          {showGroups && !!optionGroups.length ? (
+            <div className={customStyles.menuSection}>
+              <div className={customStyles.groupHeader}>Fixed roles</div>
+              <div className={styles.optionBody}>
+                {optionGroups.map((option, i) => (
+                  <RoleMenuGroupOption
+                    data={option}
+                    key={i}
+                    isSelected={groupSelected(option.value) || groupPartiallySelected(option.value)}
+                    partiallySelected={groupPartiallySelected(option.value)}
+                    onChange={onGroupChange}
+                    onClick={onMenuGroupClick}
+                  />
+                ))}
+              </div>
+            </div>
+          ) : (
+            <div className={customStyles.menuSection}>
+              <div className={customStyles.groupHeader}>Fixed roles</div>
+              <div className={styles.optionBody}>
+                {fixedRoles.map((option, i) => (
+                  <RoleMenuOption
+                    data={option}
+                    key={i}
+                    isSelected={
+                      !!(
+                        (option.uid && !!selectedOptions.find((opt) => opt.uid === option.uid)) ||
+                        !!selectedBuiltInRoles.find((role) => role.uid === option.uid)
+                      )
+                    }
+                    disabled={!!(option.uid && !!selectedBuiltInRoles.find((role) => role.uid === option.uid))}
+                    onChange={onChange}
+                    // onClick={onMenuGroupClick}
+                    hideDescription
+                  />
+                ))}
+              </div>
+            </div>
+          )}
+          {!!customRoles.length && (
```

Comment:
Both fixed and custom roles are rendered the same way, should they be combined into a single array and rendered together?